spec/features/staff/users_can_edit_a_transaction_spec.rb

```diff
@@ -67,4 +67,36 @@
       end
     end
   end
+
+  context "when signed in as a delivery partner" do
+    let(:user) { create(:delivery_partner_user) }
+    let(:activity) { create(:project_activity, organisation: user.organisation) }
+    let(:transaction) { create(:transaction, parent_activity: activity) }
+    let(:report) { create(:report, organisation: activity.organisation, fund: activity.associated_fund) }
+
+    before { authenticate!(user: user) }
+
+    context "when the transaction can be edited" do
+      before do
+        transaction.update(report: report)
+        report.update(state: :active)
+      end
+
+      scenario "shows the edit link" do
+        visit organisation_activity_path(activity.organisation, activity)
+
+        expect(page).to have_link t("default.link.edit"), href: edit_activity_transaction_path(activity, transaction)
+      end
+    end
+
+    context "when the transaction cannot be edited" do
+      before { report.update(state: :active) }
```

Comment:
Is the reason this transaction cannot be edited because the report is not the one associated to the activity?

Reply:
Yep, it's not super clear here is it – any suggestions on how I might improve this?

Reply:
Not sure. Maybe linking the transaction to a complete different report on the `before` block? But if feels redundant to do that because at the beginning of the bigger context you set up the variables, including the `report` one, and you link that to the correct fund and organisation. So I should've assumed that the reason why this transaction is not editable is because you're not updating the transaction to reflect this report as the report the transaction belongs to (if that makes sense?). In summary, I think it's clear enough as it is, and it was just me that I'm too junior 😅

Reply:
Don't say that! It's a fair comment and has nothing to do with your job title! 

I think when I wrote this one I only saw it as 'testing if the edit link shows up' rather than 'testing if this user can edit this thing in this state' (as that is taken care of by the policy and other specs. So I figured as long as this spec shows that the transaction cannot be edited, it doesn't have to be as clear why.

Reply:
That makes sense. Thanks for clarifying!